### DISCUSSION TITLE
Allow specification of processor names

### DISCUSSION THREAD

**Discussion by apexearth** [subsquid/squid-sdk#206] [Ideas]
Would be nice for those who are using multiple processors to be able to assign each processor a name so that logging is more distinguishable.

Currently able to achieve this by doing the following:

```typescript
const hackableProcessor = processor as any
hackableProcessor.getLogger = () => createLogger(`sqd:processor-${name}`)
```

**Comment by eldargab** [comment]
Usually multiple processors are run as a separately launched processes and that's how one distinguishes logs.

You can use some process manager to automate process launching and monitoring. Process managers usually provide nice interface to view logs from multiple processes as a single tagged stream. [sqd run](https://docs.subsquid.io/squid-cli/run/) is one such option.

As of your suggestion, in general logs come from multiple modules, so adjusting namespace of a `Processor` logger doesn't complete the story.
